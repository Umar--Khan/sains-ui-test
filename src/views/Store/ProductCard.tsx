import Card from 'src/shared/Card/Card';
import { Button } from 'src/shared/Buttons';
import { Product } from './Store.d';

export interface ProductCardProps {
  product: Product;
  handleOnClick: (arg: Product) => void;
}

const ProductCard = ({ product, handleOnClick, ...rest }: ProductCardProps) => (
  <Card key={product.sku} {...rest}>
    <img src={product.image} alt={product.title} />
    <p>
      {product.title}, &#163;{product.price}
    </p>
    {/*<label htmlFor='qty'>Qty:</label>*/}
    {/*<Select*/}
    {/*  name='qty'*/}
    {/*  options={quantityOptions}*/}
    {/*  onChange={() => {*/}
    {/*    console.log();*/}
    {/*  }}*/}
    {/*  value={'1'}*/}
    {/*/>*/}
    <Button onClick={() => handleOnClick(product)}>Add to basket</Button>
  </Card>
);

export default ProductCard;
