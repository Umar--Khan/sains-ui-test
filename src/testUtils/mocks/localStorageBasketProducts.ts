import { BasketProduct } from 'src/views/Store/Store.d';

export const mockedLocalStorageBasketProducts = [
  {
    productDetail: {
      'productId': '5493179',
      'sku': '549/3179',
      'title': 'Russell Hobbs Pennine Illuminating S Steel Kettle 20444',
      'price': 24.99,
      'image': 'https://media.4rgos.it/s/Argos/9805244_R_SET?w=110&h=130',
    },
    quantity: 5,
  },
  {
    productDetail: {
      'productId': '4251824',
      'sku': '425/1824',
      'title': 'Challenge White Desk Fan - 12 Inch',
      'price': 14.99,
      'image': 'https://media.4rgos.it/s/Argos/4251824_R_SET?w=110&h=130',
    },
    quantity: 1,
  },
  {
    productDetail: {
      'productId': 'tuc132404445',
      'sku': 'tuc132404445',
      'title': 'Plain Vest Top 12 BLACK',
      'price': 16,
      'image': 'https://media.4rgos.it/s/Argos/tuc132404445_R_SET?w=110&h=130',
    },
    quantity: 2,
  },
] as BasketProduct[];
