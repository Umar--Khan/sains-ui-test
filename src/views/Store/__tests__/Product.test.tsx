import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';

import ProductCard, { ProductCardProps } from '../ProductCard';
import { mockedGetProducts } from 'src/testUtils/mocks/getProducts';

describe('<ProductCard />', () => {
  const props = {
    product: mockedGetProducts[0],
    handleOnClick: jest.fn(),
  } as ProductCardProps;

  it('Should render correctly', () => {
    const component = renderer.create(<ProductCard {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Should call the onClick prop', () => {
    render(<ProductCard {...props} />);
    fireEvent.click(screen.getByText('Add to basket'));
    expect(props.handleOnClick).toHaveBeenCalledTimes(1);
    expect(props.handleOnClick).toHaveBeenCalledWith(mockedGetProducts[0]);
  });
});
