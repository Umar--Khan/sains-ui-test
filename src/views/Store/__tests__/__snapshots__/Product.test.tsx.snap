// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> Should render correctly 1`] = `
<div
  className="sc-bdfBwQ bvTWIO"
>
  <img
    alt="Russell Hobbs Pennine Illuminating S Steel Kettle 20444"
    src="https://media.4rgos.it/s/Argos/9805244_R_SET?w=110&h=130"
  />
  <p>
    Russell Hobbs Pennine Illuminating S Steel Kettle 20444
    , Â£
    24.99
  </p>
  <button
    className="sc-gsTCUz fSmUuA"
    onClick={[Function]}
  >
    Add to basket
  </button>
</div>
`;
